geminidr/core/primitives_stack.py

```diff
@@ -153,25 +157,16 @@ def stackFrames(self, adinputs=None, **params):
 
         # We will determine the average gain from the input AstroData
         # objects and add in quadrature the read noise
-        gains = [ad.gain() for ad in adinputs]
-        read_noises = [ad.read_noise() for ad in adinputs]
+        gain_list = [ad.gain() for ad in adinputs]
+        rn_list = [ad.read_noise() for ad in adinputs]
 
         # Determine whether we can construct these averages
-        process_gain = all(g is not None and not isinstance(g, list)
-                           for gain in gains for g in gain)
-        process_rn = all(rn is not None and not isinstance(rn, list)
-                         for read_noise in read_noises for rn in read_noise)
+        process_gain = not any(g is None for g in flatten(gain_list))
+        process_rn = not any(rn is None for rn in flatten(rn_list))
```

Comment:
just being pedantic, but why not
`if not None in flatten(rn_list):`

Reply:
Good idea. This code evolved and I missed the simplification.